sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2330,7 +2330,7 @@ class DataFrameSuite extends QueryTest
           new File(uuid, "file3").getAbsolutePath,
           uuid).rdd
       }
-      assert(e.getMessage.startsWith("Path does not exist"))
+      assert(e.getErrorClass == "PATH_NOT_FOUND")
```

Comment:
The current `checkError` seems not allow pattern matching for `parameter` checking.

Let me just check the errorClass for now.

Reply:
nit: Can the empty `finally` block in this case be cleaned together in this pr?

Reply:
> The current checkError seems not allow pattern matching for parameter checking.

It does, see https://github.com/apache/spark/blob/74668e2bf14760dbc60509f7736f410c09084697/core/src/test/scala/org/apache/spark/SparkFunSuite.scala#L310

Please, use `checkError` here. Just set `matchPVals` to `true`.